eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/NoImplInPublicAPI.java

```diff
@@ -50,7 +52,7 @@ public void visitToken(DetailAST ast) {
         if (ast.getType() == TokenTypes.PACKAGE_DEF) {
             String packageName = FullIdent.createFullIdent(ast.findFirstToken(TokenTypes.DOT)).getText();
             this.isTrackTwo = packageName.startsWith(COM_AZURE);
-            this.isImplPackage = packageName.endsWith(DOT_IMPLEMENTATION);
+            this.isImplPackage = Pattern.matches(IMPL_PACKAGE_REGULAR_EXPR, packageName);
```

Comment:
Rather than a regular expression, just check if the packageName.contains(".implementation") - far easier to read and far less likely to introduce bugs.